src/common/log/log.hpp

```diff
@@ -0,0 +1,249 @@
+/*
+ * Copyright (C) 2023 Renesas Electronics Corporation.
+ * Copyright (C) 2023 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef LOG_COMMON_HPP_
+#define LOG_COMMON_HPP_
+
+#include <cstdio>
+#include <cstring>
+
+#include "config.hpp"
+#include "utils/stringer.hpp"
+#include "utils/utils.hpp"
+
+/**
+ * Helper macro to display debug log.
+ */
+#if CONFIG_LOG_LEVEL >= CONFIG_LOG_LEVEL_DEBUG
+#define LOG_MODULE_DBG(module) aos::Log(module, LogLevelType::eDebug)
```

Comment:
Perhaps we could consider including information about the function using the `__FUNCTION__` macro?

Reply:
I guess it is overhead. The developer always can find the line by log payload. For others this information is useless. Would be useful in assert.

Reply:
It's up to you, but what I meant was the following use case: when you use `LOG_DBG() << "first" << 1 << "second"`, the output is `first 1 second`. If you try to find this string later, you won't be able to locate it easily. Therefore, it would be better to have a function that makes it easier to identify where the log is coming from.

Reply:
We have module at least for now. Let's go without function name for now. Can be added later if we experience problems.

Reply:
ok, agreed